### DISCUSSION TITLE
GitHub page

### DISCUSSION THREAD

**Discussion by Ksmart234** [community/community#66690] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I hosted my page on GitHub, but only my loader assest is showing...
![Screenshot_20230910-160947](https://github.com/community/community/assets/100024079/e795e49f-d872-497d-b487-7ab5fbd6fe2a)
This is the link to my repo: https://github.com/Ksmart234/Ksmart234.github.io/deployments/github-pages

**Comment by seyLu** [comment]
#### Hi @Ksmart234,

When using a static site generator, in this case you are using vite, it is recommended to build and deploy to `gh-pages` branch instead of `main`. This ensures the links are working and assets are properly loaded.

You can try any of the ff:
- [Vite Deploy Demo](https://github.com/sitek94/vite-deploy-demo)
- [Educative.io: Deploying Vite to Github Pages](https://www.educative.io/answers/deploying-vite-project-to-github-pages)

If that doesn't work, then try gh-pages npm package:
- [Deploying Vite app to Github Pages](https://medium.com/@aishwaryaparab1/deploying-vite-deploying-vite-app-to-github-pages-166fff40ffd3)
- [Auto-publish Vite proj to Github Pages](https://briangrinstead.com/blog/publishing-vite-project-to-github-pages/)

You can read more at:
- [Github Official Docs: Configuring a Publishing source for Github Pages site](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site)
